### PR TITLE
[Wip] Add handling of the 'x5u' jwt attribute

### PR DESCRIPTION
**Pull request by fermaem**
:warning: This is still a work in progress... but intends to eventually add basic support for the **x5u** jwt attribute.

Although this works on my machine, I'd like to add some tests.

My main issue is that I'm currently unable to correctly run the tests suite. Any help would be appreciated.

To that purpose, I've added a `.travis.yml` build recipe. Would you be able to help me fix it, I could move forward with adding some tests. :pray: 

Current build logs can be read at https://travis-ci.org/axa-assistance/lua-resty-jwt/

Below an excerpt of the first failure:

```
[...]

upgraded, 0 newly installed, 0 to remove and 16 not upgraded.
 ---> a48e3ebbbaf8
Removing intermediate container f71f2903024c
Step 5 : RUN ln -s /usr/nginx/sbin/nginx /usr/sbin/nginx
 ---> Running in d397fef47975
 ---> 907320f8f658
Removing intermediate container d397fef47975
Step 6 : COPY . /lua-resty-jwt/
 ---> 14ccab806f5f
Removing intermediate container d4f80b80fdda
Successfully built 14ccab806f5f
$ docker run --rm -it -v `pwd`:/lua-resty-jwt lua-resty-jwt make test
PATH=/usr/local/openresty/nginx/sbin:$PATH prove -I../test-nginx/lib -r t
t/load-verify.t .. 1/84 
#   Failed test 'TEST 11: JWT simple with default leeway and invalid exp - response_body - response is expected (repeated req 0, req 0)'
#   at /usr/local/share/perl/5.20.2/Test/Nginx/Socket.pm line 1277.
# @@ -1,2 +1,2 @@
#  false
# -jwt token expired at: Thu, 01 Jan 1970 00:00:00 GMT
# +invalid jwt string
#   Failed test 'TEST 11: JWT simple with default leeway and invalid exp - response_body - response is expected (repeated req 1, req 0)'
#   at /usr/local/share/perl/5.20.2/Test/Nginx/Socket.pm line 1277.
# @@ -1,2 +1,2 @@
#  false
# -jwt token expired at: Thu, 01 Jan 1970 00:00:00 GMT
# +invalid jwt string
```

### CODE REVIEW

**Comment by fermaem** [comment]
So... Looks like some of the tests are passing when others are don't

```
Test Summary Report
-------------------
t/load-verify.t (Wstat: 5632 Tests: 84 Failed: 22)
  Failed tests:  2, 5, 8, 11, 20, 23, 31-36, 38, 41, 50
                53, 56, 59, 68, 71, 74, 77
  Non-zero exit status: 22
t/sign-verify.t (Wstat: 1536 Tests: 90 Failed: 6)
  Failed tests:  2, 5, 62, 65, 68, 71
  Non-zero exit status: 6
```

@SkyLothar Is there any configuration setting I should tweak to have them all pass?

**Comment by SkyLothar** [comment]
hmm... thank you for the PR.
I'll check those tests later today.

**Comment by SkyLothar** [comment]
sorry for the trouble.
some of previous PRs broke the code and i didn't know.

it's okay now, thank you for the contributing!

**Comment by nulltoken** [comment]
```
t/load-verify.t .. ok
t/sign-verify.t .. ok
All tests successful.
Files=2, Tests=180,  5 wallclock secs ( 0.04 usr  0.01 sys +  0.32 cusr  0.09 csys =  0.46 CPU)
Result: PASS
```

@SkyLothar I've rebased the PR and added a basic test that leverages the new use case (which seems to make Travis happy :tada:).

I've also taken the liberty to add some new certs to cover the whole pipeline of validation. Would you prefer to use a different ones, please share them with me and I'll update the PR accordingly.

I think the code is now ready for a first pass of code review.

**Comment by fermaem** [comment]
```
t/load-verify.t .. ok
t/sign-verify.t .. ok
All tests successful.
Files=2, Tests=180,  5 wallclock secs ( 0.04 usr  0.01 sys +  0.32 cusr  0.09 csys =  0.46 CPU)
Result: PASS
```

@SkyLothar I've rebased the PR and added a basic test that leverages the new use case (which seems to make Travis happy :tada:).

I've also taken the liberty to add some new certs to cover the whole pipeline of validation. Would you prefer to use a different ones, please share them with me and I'll update the PR accordingly.

I think the code is now ready for a first pass of code review.

File: Dockerfile [hunk 1]
```diff
@@ -35,3 +35,5 @@ RUN apt-get update && \
 RUN ln -s /usr/nginx/sbin/nginx /usr/sbin/nginx
 
 COPY . /lua-resty-jwt/
+
+RUN ls /lua-resty-jwt/testcerts
```
Comment (SkyLothar) [inline Dockerfile]: hmm... you want to make sure the directory exists?

that's okay. i think i'll write a better dockerfile.
this one is too s....l.....o....w.....

**Comment by SkyLothar** [comment]
wonderful! 

:+1: :shipit: 

i'll make a new release in few days!

**Comment by nulltoken** [comment]
> i'll make a new release in few days!

Any chance this lib could be also made available through LuaRocks?

**Comment by SkyLothar** [comment]
@nulltoken and #15 
Great Idea! :tada: 
I'll publish this lib to LuaRocks as you suggested.

**Pull request merged**